src/Domain/Creators/AlignedLattice.cpp

```diff
@@ -3,6 +3,10 @@
 
 #include "Domain/Creators/AlignedLattice.hpp"
 
+#include "Domain/CoordinateMaps/CoordinateMap.hpp"
```

Comment:
I'm a bit puzzled by all the extra includes needed in this commit; I don't see what they have to do with registration of maps_list of TimeDependences

Reply:
I needed them in the tests to get linking working with GCC7 release. I'm of the opinion that the domain creators should include the coordinate maps that they use, but I'm fine removing them if you prefer.

Reply:
Thinking more about this, I'm confused that the required code isn't instantiated anywhere. I'll think about this a bit more and try to figure it out, then report back.

Reply:
Okay, I looked at this quite a bit and I half understand what's going on.
- The includes in the non-test files are not needed, I've removed them in a fixup
- The includes in the test files are needed for a few different reasons (though I don't understand them all).
  1. The compiler needs to generate instantiations for calling the `_impl` member functions of `CoordinateMap`. I don't know why the linker can't find them elsewhere when the main executables seem to be fine. 
  2. The tests pass one of the maps by rvalue reference and the other by const lvalue reference to `make_coordinate_map_base`, while the code that creates the maps in the domain creator I think does 2 lvalue or 2 rvalue references (I didn't bother digging through it). This is a legitimate reason to need the `CoordinateMap.tpp` include in the test files.
  3. The linker cannot find pup functions for the `ProductOf?Maps`. Again, I don't understand why the executables are fine but the tests aren't. This makes no sense to me.

  For 2., the answer is including the `tpp`. For 1 and 3 I've written some macros to auto-instantiate all the necessary code. I'm not sure of a good way to test that all the explicit instantiations are generated because as I mentioned some includes for `tpp`s are actually needed by the tests. I can add a commit that adds the explicit instantiation macros and uses them for the cases encountered here if desired.